features/dcs_failsafe_mode.feature

```diff
@@ -0,0 +1,72 @@
+Feature: dcs failsafe mode
+  We should check the basic dcs failsafe mode functioning
+
+  Scenario: check failsafe mode can be successfully enabled
+    Given I start postgres0
+    And postgres0 is a leader after 10 seconds
+    And I sleep for 3 seconds
+    When I issue a PATCH request to http://127.0.0.1:8008/config with {"loop_wait": 2, "ttl": 20, "retry_timeout": 5, "failsafe_mode": true}
+    Then I receive a response code 200
+    And I sleep for 3 seconds
+    When I issue a GET request to http://127.0.0.1:8008/failsafe
+    Then I receive a response code 200
+    And I receive a response postgres0 http://127.0.0.1:8008/patroni
+    When I issue a PATCH request to http://127.0.0.1:8008/config with {"postgresql": {"parameters": {"wal_level": "logical"}}}
+    Then I receive a response code 200
+    When I issue a PATCH request to http://127.0.0.1:8008/config with {"slots": {"dcs_slot_0": {"type": "logical", "database": "postgres", "plugin": "test_decoding"}}}
+    Then I receive a response code 200
+
+  @dcs-failsafe
+  Scenario: check one-node cluster is functioning while DCS is down
+    Given DCS is down
+    Then postgres0 role is the primary after 20 seconds
+
+  @dcs-failsafe
+  Scenario: check new replica isn't promoted when leader is down and DCS is up
+    When I do a backup of postgres0
+    And I shut down postgres0
+    And DCS is up
+    When I start postgres1 in a cluster batman from backup with no_master
+    And I sleep for 2 seconds
+    Then postgres1 role is the replica after 12 seconds
+
+  @dcs-failsafe
+  Scenario: check leader and replica are both in /failsafe key after leader is back
+    Given I start postgres0
+    Then "members/postgres0" key in DCS has state=running after 10 seconds
+    And "members/postgres1" key in DCS has state=running after 2 seconds
+    When I sleep for 3 seconds
+    And I issue a GET request to http://127.0.0.1:8009/failsafe
+    Then I receive a response code 200
+    And I receive a response postgres0 http://127.0.0.1:8008/patroni
+    And I receive a response postgres1 http://127.0.0.1:8009/patroni
+
+  @dcs-failsafe
+  @slot-advance
+  Scenario: check leader and replica are functioning while DCS is down
+    Given logical slot dcs_slot_0 is in sync between postgres0 and postgres1 after 10 seconds
+    And DCS is down
+    And I sleep for 12 seconds
+    Then postgres0 role is the primary after 2 seconds
+    And postgres1 role is the replica after 2 seconds
+    And replication works from postgres0 to postgres1 after 10 seconds
+    And I get all changes from logical slot dcs_slot_0 on postgres0
+    And logical slot dcs_slot_0 is in sync between postgres0 and postgres1 after 10 seconds
+
+ @dcs-failsafe
+  Scenario: check three-node cluster is functioning while DCS is down
+    Given DCS is up
+    And I start postgres2
+    Then "members/postgres2" key in DCS has state=running after 10 seconds
```

Comment:
@CyberDem0n shouldn't we also check content of `GET http://127.0.0.1:8009/failsafe` here...?

Reply:
Do you expect it not to work despite working at https://github.com/CyberDem0n/patroni/pull/4/files#diff-e200798e6805f00fc721233822d3b238e7f9eb81187ec5febd68df6f79747d6cR39-R42?

Reply:
Can't remember the point I wrote this yesterday. Seems it wasn't that relevant anyway...